### PR TITLE
Tdd

### PR DESCRIPTION
**Pull request by aron-net**
:heavy_check_mark:  A class was created called Solver.
:heavy_check_mark:  A method was created called factorial 
- That takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. 
- The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. 
- This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
:heavy_check_mark:  A method was created called reverse 
-That takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
:heavy_check_mark:  A method was created called fizzbuzz 
- That takes one argument, an integer N, and returns a string. 
- The returned string is constructed following these rules:
- When N is divisible by 3, return "fizz".
- When N is divisible by 5, return "buzz".
- When N is divisible by 3 and 5, return "fizzbuzz".
- Any other case, return N as a string (e.g. say N is 7 then return "7").

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  before(:each) do
+    @solver = Solver.new
+  end
+
+  context 'factorial method' do
+    it 'take N and return factorial for that number' do
+      result = @solver.factorial(0)
+      expect(result).to eq 1
+    end
+  end
```
Comment (codepantha) [inline spec/solver_spec.rb]: - [ ] For the factorial, please write a test that can handle a negative value which raises an exception as an expectation 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🎯 
✔️ Descriptive PR
✔️ Wrote tests with RSpec
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please reference this [template](https://github.com/microverseinc/readme-template) to make your README more professional. Ensure to have all the headings 👍 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (aron-net) [reply spec/solver_spec.rb]: Already implemented it! Thanks for the feedback! :smile:

**Review by DammyShittu — COMMENTED** [review]
Hi @aron-helu @Eternalgratis,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Fix your linter errors

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Ismailco — APPROVED** [review]
### Hi @aron-helu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by aron-net** [comment]
Thank you for reviewing.

**Pull request merged**